src/effects/presets/effectchainpresetmanager.cpp

```diff
@@ -788,13 +828,40 @@ EffectsXmlData EffectChainPresetManager::readEffectsXml(
         }
     }
 
-    return EffectsXmlData{quickEffectPresets, standardEffectChainPresets};
+    return EffectsXmlData{
+            eqEffectManifests, quickEffectPresets, standardEffectChainPresets, mainEqPreset};
 }
 
-EffectChainPresetPointer EffectChainPresetManager::readEffectsXmlSingleDeck(
+EffectXmlDataSingleDeck EffectChainPresetManager::readEffectsXmlSingleDeck(
         const QDomDocument& doc, const QString& deckString) {
     QDomElement root = doc.documentElement();
 
+    // EQ effect
+    auto pEqEffect = getDefaultEqEffect();
+
+    // Read id of last loaded EQ effect
+    QDomElement eqEffectsElement =
+            XmlParse::selectElement(root, EffectXml::kEqualizerEffects);
+    QDomNodeList eqEffectNodeList =
+            eqEffectsElement.elementsByTagName(
+                    EffectXml::kEffectId);
+    for (int i = 0; i < eqEffectNodeList.count(); ++i) {
+        QDomElement effectIdElement = eqEffectNodeList.at(i).toElement();
```

Comment:
for (const auto& node : eqEffectNodeList) {
    QDomElement effectIdElement = node.toElement();

Reply:
I know this is clunky, but QDomNodeList does not provide iterators unfortunately.